src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -13,18 +13,27 @@
 (def ServiceStatusDetailLevel
   (schema/enum :critical :info :debug))
 
+(def StatusCallbackResponse
+  {:is-running schema/Bool
+   :status schema/Any})
+
 (def ServiceInfo
   {:service-version schema/Str
    :service-status-version schema/Int
-   ;; TODO: specify output of callback function in this schema?
-   :status-fn (schema/make-fn-schema ServiceStatusDetailLevel schema/Any)})
+   ;; Note that while this specifies the input and output for the status
+   ;; function for each service, it does not actually validate these
+   :status-fn (schema/make-fn-schema StatusCallbackResponse ServiceStatusDetailLevel)})
 
 (def ServicesInfo
   {schema/Str [ServiceInfo]})
 
+;; this is what gets returned in the HTTP response as json, and thus should
+;; use underscores rather than hyphens
 (def ServiceStatus
-  {:service-version schema/Str
-   :service-status-version schema/Int
+  {:service_version schema/Str
+   :service_status_version schema/Int
+   :is_running (schema/enum true false :unknown)
+   :detail_level ServiceStatusDetailLevel
    :status schema/Any})
```

Comment:
Ergh, so the problem with reusing `StatusCallbackResponse` in this is that this uses underscores (because ultimately it's going to be JSON output), while that has hyphens. We could just use underscores everywhere, I guess? Or just keep them separate.